lumeer-core/src/main/java/io/lumeer/core/auth/PermissionsChecker.java

```diff
@@ -445,6 +445,46 @@ public void checkDocumentLimits(final List<Document> documents) {
       }
    }
 
+   public void checkRulesLimit(final Collection collection) {
+      if (skipLimits()) {
+         return;
+      }
+
+      final ServiceLimits limits = paymentFacade.getCurrentServiceLimits(workspaceKeeper.getOrganization().get());
+      if (limits.getRulesPerCollection() != 0 && collection.getRules().size() > limits.getRulesPerCollection()) {
+         throw new ServiceLimitsExceededException(collection.getRules(), limits.getRulesPerCollection());
+      }
+
+   }
+
+   public void checkFunctionsLimit(final Collection collection) {
+      if (skipLimits()) {
+         return;
+      }
+
+      final ServiceLimits limits = paymentFacade.getCurrentServiceLimits(workspaceKeeper.getOrganization().get());
+      if (limits.getFunctionsPerCollection() != 0) {
+         long functions = collection.getAttributes().stream().filter(attribute -> attribute.getFunction() != null).count();
```

Comment:
It's not enought to test it like this: `attribute.getFunction() != null`. We talked about that earlier. You have to test if javascript inside function is empty or not.